janus_client/src/lib.rs

```diff
@@ -57,9 +57,18 @@ impl ClientParameters {
     /// Creates a new set of client task parameters.
     pub fn new(
         task_id: TaskId,
-        aggregator_endpoints: Vec<Url>,
+        mut aggregator_endpoints: Vec<Url>,
         min_batch_duration: Duration,
     ) -> Self {
+        // Ensure provided aggregator endpoints end with a slash, as we will be joining additional
+        // path segments into these endpoints & the Url::join implementation is persnickety about
+        // the slash at the end of the path.
+        for url in &mut aggregator_endpoints {
+            if !url.as_str().ends_with('/') {
+                url.set_path(&format!("{}/", url.path()));
+            }
+        }
```

Comment:
Meganit: we could put an extension trait on `Url` in `janus_core` that defines a method `Url::normalize` that does this. Then again this code probably won't change so we can live with a little copy-pasta.